classes/converter.php

```diff
@@ -67,6 +67,10 @@ public function convert_moodle_url_to_pdf(moodle_url $url, string $key, string $
         // When implemented, the target URL must be passed through the proxy page (index.php) as a parameter
         // along with the access key, in order to validate user login.
         // {@see \tool_pdfpages\helper::get_proxy_url}.
+
+        // All converters MUST implement `destroy session` following conversion, in order to prevent improper use
+        // of the session for further actions.
+        $this->destroy_session();
```

Comment:
This will not guarantee that all converters will do Also if you try to shut down session will you be still able to print out the stored file? 
Maybe add this to a shutdown handler if that does make sense here? 

Or you can have convert_moodle_url_to_pdf as a final method and call another abstract method inside to actually get the file. Then you can be sure that you close session in convert_moodle_url_to_pdf?

Reply:
I have moved the conversion to PDF to a separate protected method to be overridden by converters and I have added the session termination to a `final public` method for the actual API, this way, the converters are responsible for getting the PDF content and MUST go through the proxy, and the session is always terminated, to prevent session hijacking.